From: Saravana Kannan <saravanak@google.com>
Date: Fri, 21 Feb 2020 00:05:09 -0800
Subject: [PATCH] driver core: Add dev_has_sync_state()
Patch-mainline: 5.5.13
References: bnc#1012628 bnc#1167245
Git-commit: ac338acf514e7b578fa9e3742ec2c292323b4c1a

commit ac338acf514e7b578fa9e3742ec2c292323b4c1a upstream.

Add an API to check if a device has sync_state support in its driver or
bus.

Signed-off-by: Saravana Kannan <saravanak@google.com>
Link: https://lore.kernel.org/r/20200221080510.197337-3-saravanak@google.com
Cc: Davide Caratti <dcaratti@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/device.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/include/linux/device.h b/include/linux/device.h
index 96ff76731e93..50d97767d8d6 100644
--- a/include/linux/device.h
+++ b/include/linux/device.h
@@ -1522,6 +1522,17 @@ static inline struct device_node *dev_of_node(struct device *dev)
 
 void driver_init(void);
 
+static inline bool dev_has_sync_state(struct device *dev)
+{
+	if (!dev)
+		return false;
+	if (dev->driver && dev->driver->sync_state)
+		return true;
+	if (dev->bus && dev->bus->sync_state)
+		return true;
+	return false;
+}
+
 /*
  * High level routines for use by the bus drivers
  */
-- 
2.26.0

